chore(build): bump MSRV to 1.81.0

overlookmotel opened the pull request:
As noted in #8929, many Oxc crates won't compile at present with our MSRV of 1.80.0, due to the liberal use of `#[expect]` attribute, which wasn't stabilized until 1.81.0.

It becomes a lint error in 1.81.0 to use `#[allow]`, so it'd seem to me pretty pointless to convert all the `#[expect]` attrs to `#[allow]` to honour our MSRV now, when we'll have to change them back to `#[expect]` again shortly.

So I propose that we bump the MSRV to 1.81.0 early instead. Next minor version of Rust is out in 2 weeks anyway.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8931?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#8931** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8931?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8931?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#8930** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8930?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=8931).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Feb 7, 8:33 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
chore(build): bump MSRV to 1.81.0 (#8931)

As noted in #8929, many Oxc crates won't compile at present with our MSRV of 1.80.0, due to the liberal use of `#[expect]` attribute, which wasn't stabilized until 1.81.0.

It becomes a lint error in 1.81.0 to use `#[allow]`, so it'd seem to me pretty pointless to convert all the `#[expect]` attrs to `#[allow]` to honour our MSRV now, when we'll have to change them back to `#[expect]` again shortly.

So I propose that we bump the MSRV to 1.81.0 early instead. Next minor version of Rust is out in 2 weeks anyway.
